spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -1795,10 +1795,19 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
           optExprWithInfo(optExpr, expr, child)
 
         case InitCap(child) =>
-          val castExpr = Cast(child, StringType)
-          val childExpr = exprToProtoInternal(castExpr, inputs)
-          val optExpr = scalarExprToProto("initcap", childExpr)
-          optExprWithInfo(optExpr, expr, castExpr)
+          if (CometConf.COMET_EXEC_EXPAND_ENABLED.get()) {
```

Comment:
It looks like the wrong config is referenced here?

```suggestion
          if (CometConf.COMET_EXEC_INITCAP_ENABLED.get()) {
```

Reply:
My fault, Thank you!